### PR TITLE
dialog window terms conditions

### PR DESCRIPTION
**Pull request by volodymyrmakarchuk-okta**
## Description:
- **What's changed?** 
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-364713](https://oktainc.atlassian.net/browse/OKTA-364713)
* [OKTA-378018](https://oktainc.atlassian.net/browse/OKTA-378018)

### Netlify deploy
https://6053c3474eb6ce0fa7bff09f--reverent-murdock-829d24.netlify.app/signup/

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/assets/css/components/_dialogWindow.scss
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/assets/css/components/_dialogWindow.scss:32]: Typo here, should be `.dialog--...`

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:230]: Remove commented out code

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:247]: Remove commented out code

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:226]: Can we move the modal opening and closing code into methods?

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:226]: if you meant to create open/close methods for `DialogWindow` component then we probably need to create that component programmatically by using `new` for example and then it will be possible to call open/close methods from outside or I didn't get you? @francoislehoux-okta

File: packages/@okta/vuepress-theme-prose/assets/css/components/_dialogWindow.scss
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/assets/css/components/_dialogWindow.scss:32]: done

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue]: done

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue]: done

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:495]: 🚧  Are we awaiting the URL schema to be finalized? We'll need to just remember to block merging until this is resolved. This comment can serve as that blocker. 🚧

**Review by shawyu-okta — CHANGES_REQUESTED** [review]
💯  for updating/adding selenium tests -- few comments, one blocker to make sure we address the social IDP URL schema before merge.

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:495]: https://oktainc.atlassian.net/browse/OKTA-368714

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:495]: Got it! That merged this morning so it will not be available until next release 2020.03.1, which should be out on 2021-03-16

Will social login still work if we use these parameters in the URL before the change is released, or do we have to hold this change until the monolith release is deployed to OK12?

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:495]: I'm not sure what should I get in case of successfully complete social auth but I see this page
Screenshot at Mar 01 18-21-24

File: packages/@okta/vuepress-site/.vuepress/enhanceApp.js
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/.vuepress/enhanceApp.js:7]: hmm, I've missed that, thank you

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue]: ^This

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue]: done

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:225]: I think I imagined this as `@click="socialButtonClickHandler('google')"` or something like that. @francoislehoux-okta any thoughts?

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:225]: Apologies for not listing this suggestion sooner

Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:225]: That was my thought yeah, my bad for not being clear enough

**Comment by albertchen-okta** [comment]
A new design is incoming for this due to updates from our Legal team: https://www.figma.com/file/fn5Qqou29vkSGcppiLN4sl/Dev-Console-Migration?node-id=997%3A3686 stay tuned

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:142]:
```suggestion
      // If country Canada or USA but state not selected, disable "continue" button.
```

**Review by francoislehoux-okta — COMMENTED** [review]
I can't remember if we had decided to add a backdrop/overlay when the modal is open, did we?

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:36]:
```suggestion
          name="State"
```

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:142]: yes, this is from the new task https://oktainc.atlassian.net/browse/OKTA-378018

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:36]: hhmm probably will be a good idea to use `region` instead, this field could be `state` or `province`

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:218]: Wondering why these 2 are inputs and not buttons

**Review by francoislehoux-okta — APPROVED** [review]

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:218]: I have a habit to use an input type button instead of button tag if it does not require some specific content inside. do you have some thought about `input type button` vs `button`

File: packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss:9]: Can we use a more descriptive name for the class? Classes should be normally named after the element it represents, rather than what style it will apply. If the margin ever changes to something else (e.g. 20px) this class name won't make sense anymore.

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:158]: nit: we should have consistent naming, either `canadianProvinces` or change above to `americaStates` (I prefer the first one)

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:166]: typo? `countriesList`

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:188]: I believe this should use `encodeURIComponent` because `encodeURI` does not encode `&` `+` and `=`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs_encodeuricomponent

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:213]: Perhaps we should consider using a library or writing a small utility that does URI construction. This can get sticky fast.

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:223]: Should the scss be in the compiled partials like the other components?

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:221]: I think using normal sentence casing "Continue with GitHub" and using CSS to `text-transform: uppercase` will be more accessible and flexible

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:447]: Looks like a bit of a fight between IDEs/Prettier autoformatting. Was this autoformatted?

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:287]: Is this right? This should be a concrete value, not a data type based on the rest of the fields around it.

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue]: Yeah it will still succeed but some parameters in the social URL will be ignored by the backend. We need to make sure the backend is actually accepting the parameters before we can show the modal to the user.

**Comment by shawyu-okta** [comment]
> I can't remember if we had decided to add a backdrop/overlay when the modal is open, did we?

@francoislehoux-okta there is a followup ticket filed: https://oktainc.atlassian.net/browse/OKTA-378018

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:158]: hm, Canada use province like the word the describe some territory and USA use state that have the same meaning

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:158]: Oh, I'm referring the first half of the variable, not `states` vs `provinces`

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:218]: Ah, missed the `type="button"` part. Not against it, they're both quite similar although one is more verbose than the other. I think people used to prefer the `input` way for backward compatibility with IE if im not mistaken.

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:158]: Shaw is referring to `canada/america` vs `canadian/american`, you currently have them mixed up.

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:158]: my bad sorry =)

File: packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss:9]: in css quite common case using a bunch of auto-generated classes for margins, paddings etc. In my case, I have only one class but I tried to follow the same idea, this class was declared on the global level because I expected that it could be used in different places of the signup page where concrete 15 margin needed. Perfectly we need to have such auto-generated classes for padding, margins, and somewhere I've seen such, we can try to use that file, what do you think?

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:213]: does it mean that we should create some helper class or mixin for URI purposes and to put this code inside?

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:447]: yes it was autoformatted with prettier in my case

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:287]: `socialUrl`  is undefined until a user clicks on signup with  google/github .

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:287]: Should it just be an empty string then?
```suggestion
      socialUrl: '',
```

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:213]: i've written this exact utility before but it can be tricky so using a library would be a good way to accomplish this too

File: packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss
Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/assets/css/layouts/_authorizationLayout.scss:9]: I've had issues with utility classes conflicting with other classes, and then having specificity resolve conflicts, which can lead to unintended side-effects in the future when the class or DOM structure changes. It works well when it's defined as a part of a broad utility styling library like Bootstrap, but IMO it's not worth trying to make it up on one's own.

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:95]: Is the `!isDisabledSocialAuth` part necessary if the input is disabled via the `:disabled` property below?

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:117]: Why create this constant instead of directly putting `""` in the places below?

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: How can the object have fields with an empty key value? Based on the `queryParams` definition above the only possible one I can see is the dynamically inferred `[regionType]` key.

File: packages/@okta/vuepress-theme-prose/global-components/SignUp.vue
Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/global-components/SignUp.vue:287]: Make sure to address this before merge

**Review by shawyu-okta — APPROVED** [review]
A few last comments to address but once resolved, you're good to :shipit: to the feature branch -- please squash merge :)

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:95]: it's a protecting mechanism if someone will want to remove disabled attr maually from dev console. Not sure about prod build but in dev env removing disabled attr will allow executing onclick  event

File: packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: yes in this case it's only one property that could be empty but logic could be changed. e.g. a new property added and in some cases it could be empty

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: By definition only one key could be empty though 🤔

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: I'm not sure that I've caught your thought, could you please describe in more detail?

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: The comment says "Remove object fields with empty "key" value." But hash key/value pairs mean only only one entry with any "key", including `""` e.g.
```js
const myObj = {};
myObj[""] = "some value";

console.log(myObj); // { '': 'some value' }

myObj[""] = "other value";

console.log(myObj); // { '': 'other value' }
```
Only one entry can have any particular key ^

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: well spotted! I haven't thought about that, thanks.
 what if to use `delete obj[""]` ?

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue:197]: Using `delete` would work -- I think overall it might be clearer to directly check if `regionType` is not empty, then add it to `queryParams` like this:
```js
const queryParams = {
        okta_AcceptedToS,
        okta_ts_AcceptedToS,
        country,
      };
if (regionType !== "") {
  queryParams[regionType] = regionName;
}
```

**Pull request merged**
